docs/articles/book/concepts/TestingConcepts.md

```diff
@@ -0,0 +1,88 @@
+
+* **F**ast: We want our tests to run as fast as possible. For example, unit tests should typically execute in no more than a second. Tests at higher levels might take slightly longer, but we want to prioritize speed.
+  * Fast tests mean as many tests as possible can be run as often as possible -- this is the source of confidence that automated tests provide.
+* **I**solated: Tests should not depend on the output of another test or on a specific order of tests or production code run. We should (theoretically) be able to run all tests at the same time.
+* **R**epeatable: The same test, when run many times, should always produce the same result.
+* **S**elf-Documenting: The test itself should tell you whether it passed or failed.
+  * By this, we mean that a test shouldn't, for example, write out to a file that a human then needs to examine. When we run the test suite, we should gain immediate confidence in whether our code passes those tests or not.
+  * Assertions are how we accomplish this in automated test libraries.
+* **T**imely: Tests should be written around the same time as production code, so that you are not just documenting what the code does, but _what you intended_ the code to do.
+  * In the case of TDD, the tests are guaranteed to be timely, because they're written _before_ the production code.
+
+## Tests Are Production Code, Too
+
+Just because you have tests in a test project doesn't mean they should be programmed in a sloppy manner. Active maintenance and refactoring of tests can pay huge dividends over the life of a project.
+
+## "Flaky" Tests
+
+Sometimes you'll hear tests referred to as "flaky" or experiencing "flakiness". Typically this means a test sometimes fails when we don't expect it to for reasons we don't fully understanding (violating the `R` in `FIRST`, stating that tests should be repeatable).
+
+Typically, we want to fix these tests so that we can trust our test suite and don't start to ignore failures as "probably just a test being flaky". We need to be able to trust what the test results well us.
+
+## DRY vs. DAMP
+
+Because tests are production code, the general programming concept of `DRY` (Don't Repeat Yourself) often comes into play with tests.
+
+However, tests have an important consideration -- they should be understandable quickly, with as little sleuthing as possible. If it serves the readability and understandability of the test, it's perfectly fine to repeat certain logic within a test rather than attempting to abstract it all away. In this way, we often say we prefer `DAMP` (Descriptive and Meaningful Phrasing) to `DRY`.
+
+Like many things, this is a balance that a team should attempt to find and actively maintain. Some parts of tests can be extracted into common methods; other abstractions might make tests too hard to understand. Be willing to re-evaluate these choices as projects evolve.
+
+## Test Coverage
```

Comment:
I like this because it surfaces something that is a natural part of how I think but wouldn't be for others. I'll consider adding a section on "what do we test?" Noting that we want to test production code and want to avoid our tests mimicking production code with a lot of logic that itself can fail.